### PR TITLE
chore: convert remaining source files to TypeScript

### PR DESCRIPTION
**Pull request by ascorbic**
### Summary

Converts the final few files to TypeScript

### Test plan

1. Check that the deploy previews work

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

image

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

File: src/helpers/files.ts [hunk 1]
```diff
@@ -1,46 +1,54 @@
...
-const matchMiddleware = (middleware, filePath) =>
+export const matchMiddleware = (middleware: Array<string>, filePath: string): string | boolean =>
   middleware?.includes('') ||
   middleware?.find(
     (middlewarePath) =>
       filePath === middlewarePath || filePath === `${middlewarePath}.html` || filePath.startsWith(`${middlewarePath}/`),
   )
 
-const matchesRedirect = (file, redirects) => {
+export const matchesRedirect = (file, redirects: Rewrites): boolean => {
```
Comment (tiffanosaurus) [inline src/helpers/files.ts:51]:
```suggestion
export const matchesRedirect = (file: string, redirects: Rewrites): boolean => {
```

File: src/helpers/files.ts [hunk 2]
```diff
@@ -63,14 +71,16 @@ const matchesRewrite = (file, rewrites) => {
...
-
 // eslint-disable-next-line max-lines-per-function
-exports.moveStaticPages = async ({ netlifyConfig, target, i18n }) => {
+export const moveStaticPages = async ({
+  netlifyConfig,
+  target,
+  i18n,
+}: {
+  netlifyConfig: NetlifyConfig
+  target: 'server' | 'serverless' | 'experimental-serverless-trace'
```
Comment (tiffanosaurus) [inline src/helpers/files.ts:81]: [sand] Could have a type for this instead

File: src/helpers/files.ts [hunk 3]
```diff
@@ -1,46 +1,54 @@
...
+    }
+  | Array<Redirect>
 
 const TEST_ROUTE = /(|\/)\[[^/]+?](\/|\.html|$)/
 
-const isDynamicRoute = (route) => TEST_ROUTE.test(route)
+export const isDynamicRoute = (route) => TEST_ROUTE.test(route)
 
-const stripLocale = (rawPath, locales = []) => {
+export const stripLocale = (rawPath, locales = []) => {
```
Comment (tiffanosaurus) [inline src/helpers/files.ts:36]:
```suggestion
export const stripLocale = (rawPath: string, locales: Array<string> = []) => {
```
Or locales could be a type

**Review by tiffanosaurus — APPROVED** [review]
💃 looks great, lil potential improvement comments

File: src/helpers/files.ts (hunk 2 above)
Comment (ascorbic) [reply src/helpers/files.ts:81]: 👏🏻 
image

**Review by tiffanosaurus — APPROVED** [review]

**Pull request merged**